external/dynamo/generalized/router.py

```
@@ -271,17 +302,64 @@ def _iat_factor(iat: str) -> float:
+                        self.engine_client.wait_for_instances(),
+                        timeout=min(remaining, 10.0),
+                    )
+                except TimeoutError:
+                    # We'll re-check instance IDs and retry with backoff until the global deadline.
+                    pass
+
+                instance_ids_raw = list(self.engine_client.instance_ids())
+                if len(instance_ids_raw) >= min_workers:
+                    try:
+                        instance_ids = [int(w) for w in instance_ids_raw]
+                    except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)